app/src/main/java/org/fossasia/openevent/activities/MainActivity.java

```diff
@@ -105,6 +140,58 @@ private void setUpToolbar() {
         }
     }
 
+    @Subscribe
+    public void tracksDownloadDone(TracksDownloadEvent event) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.content_frame, new TracksFragment()).commit();
+
+    }
+
+    @Subscribe
+    public void EventsDownloadDone(EventDownloadEvent event) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.content_frame, new SponsorsFragment()).commit();
+
+    }
+
+    @Subscribe
+    public void sponsorsDownloadDone(SponsorDownloadEvent event) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        fragmentManager.beginTransaction()
+                .replace(R.id.content_frame, new SponsorsFragment()).commit();
+
+    }
+
+    @Subscribe
+    public void microlocationsDownloadDone(TracksDownloadEvent event) {
```

Comment:
Same function argument but different name, how will Otto differentiate? Both will get executed, in sequence which is probably not what you want?

![image](https://imgflip.com/s/meme/Jackie-Chan-WTF.jpg)